### ISSUE TITLE
Unresponsive UI on link click when using getStaticProps with fallback: true

### ISSUE THREAD

**Issue by timwachsener** [vercel/next.js#14590]
# Bug report

## Describe the bug

When using lazy statically generated pages with `fallback: true` clicking a link doesn't trigger `routeChangeStart` on the router until the props are generated on the server side (when the page wasn't accessed before). This leads to a few moments of "nothing happening".

## To Reproduce

Use an empty paths array with `fallback: true` (this is actually a production code example because we have > 100k pages and pre-generating them would take way too long).

```
export async function getStaticPaths() {
  return { paths: [], fallback: true };
}
```

## Expected behavior

Router should fire `routeChangeStart` (or another event) directly on link click when the props are requested from the server and not until when they are generated. This would allow to display a loading bar and inform the user that something is happening.

## System information

- Version of Next.js: ^9.4.4
- Version of Node.js: v13.8.0

**Comment by timneutkens** [comment]
Please add a complete reproduction.

**Comment by Visakeswaran** [comment]
I using **next 9.0.2** and facing a similar issue. I am routing to a dynamic route having around 12 api calls inside **getInitialProps()**, using **next/link**. when i trigger routing, the page becomes unresponsive until all the api fetching is done. I have referred to the with-loading example and also disabled prefetch for link, but no good.
![ezgif com-video-to-gif](https://user-images.githubusercontent.com/17028188/89519231-a5fef700-d7f9-11ea-9bcf-4c1a7239dc8b.gif)

**Comment by ijjk** [comment]
Hi, this has been updated in the latest version of Next.js `v11.1.2` and `routeChangeStart` is fired before the route starts loading, please update and give it a try!

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
